### PR TITLE
Add ElementId and test

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

File: tests/Unit/Domain/Test_ElementId.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include "Domain/ElementId.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+TEST_CASE("Unit.Domain.ElementId", "[Domain][Unit]") {
+  // Test default constructor:
+  ElementId<3> test_id;
+  CHECK(test_id.block_id() == std::numeric_limits<size_t>::max());
+  CHECK(test_id.segment_ids() == make_array<3>(SegmentId()));
+  static_cast<void>(test_id);
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_ElementId.cpp]: don't need this cast since it's being used

File: src/Domain/ElementId.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  p | segment_ids_;
+}
+
+template <size_t VolumeDim>
+std::ostream& operator<<(std::ostream& os, const ElementId<VolumeDim>& id) {
+  os << "[B" << id.block_id() << ',' << id.segment_ids() << ']';
+  return os;
+}
+
+namespace std {
```
Comment (nilsdeppe) [inline src/Domain/ElementId.cpp]: Probably want to ignore this in LCOV

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@gsb76  It turns that GitHub's interactive edits for merge conflict resolution is not as nice as I'd hope... Could you do a rebase locally and then force push over the edit on GH?

File: tests/Unit/Domain/Test_ElementId.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Test equality operator:
+  ElementId<3> element_one(1);
+  ElementId<3> element_two(1);
+  ElementId<3> element_three(3);
+  ElementId<3> element_four(4);
+  CHECK(element_one == element_two);
+  CHECK(element_two != element_three);
+  CHECK(element_two != element_four);
+  CHECK(element_three != block_2_3d);
```
Comment (wthrowe) [inline tests/Unit/Domain/Test_ElementId.cpp:43]: Check != for two different elements in the same block.

**Review by osheamonn — APPROVED** [review]

**Pull request merged**